api/JobHandler.py

```diff
@@ -16,11 +16,49 @@ def addJob(data):
 
     idLock.acquire()
     job = {'status': 'New', 'id': jobIds, 'data': data}
-    jobIds = jobIds + 1
     idLock.release()
 
     jobLock.acquire()
-    jobs.append(job)
+    exists = False
+    if len(jobs) < 1:
+        idLock.acquire()
+        jobIds = jobIds + 1
+        idLock.release()
+        jobs.append(job)
+    else:
+        for currentJob in jobs:
+            currentData = currentJob['data']
```

Comment:
why implement your own locking system (which may not work) when you could use something which is guaranteed to work such as in Berkeley DB??? docs here http://pybsddb.sourceforge.net/bsddb3.html

Reply:
I already had a filesystem solution in place so I was continuing to build on it. My primary goal was to get the system to a point where it worked, and I believe this PR is the first which does. Looking into Berkeley DB now, it has a lot of the features which I have already implemented and I did waste time implementing them due to being tunnel visioned on getting something working. I will begin investigating implementing Berkeley DB as this would be the best long term solution.

Reply:
that is fine for now, but please think about how to migrate to bdb